test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +90,9 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (inputCommand.equals("add") || inputCommand.equals("delete") || inputCommand.equals("clear")) {
```

Comment:
It's not a good practice (and in fact, contrary to SE principles) to have "magic strings" or hardcoded strings. What if, down the line, we add a new command? This would need to change every time that happens. One alternative is to actually call the `isMutating` method to determine if the command is indeed mutating.

Furthermore, this test only checks if a mutating command saves the changes to file. What about commands that are not supposed to mutate the address book? The test should cover both cases.